src/collision.cpp

```diff
@@ -405,6 +394,18 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 				}
 			}
 		}
+#ifndef SERVER
+		if (self && c_env) {
+			LocalPlayer *lplayer = c_env->getLocalPlayer();
+			if (lplayer->getParent() == nullptr) {
+				aabb3f lplayer_collisionbox = lplayer->getCollisionbox();
+				v3f lplayer_pos = lplayer->getPosition();
+				lplayer_collisionbox.MinEdge += lplayer_pos;
+				lplayer_collisionbox.MaxEdge += lplayer_pos;
+				cinfo.emplace_back(false, true, 0, v3s16(), lplayer_collisionbox);
```

Comment:
So how does the player not collide with their own collision box?

Reply:
Here: 398 `if (self && c_env)`
Localplayer doesn't set the self so took advantage of that.